DFC.ServiceTaxonomy.ApiFunction/Function/Execute.cs

```diff
@@ -0,0 +1,166 @@
+using DFC.ServiceTaxonomy.ApiFunction.Helpers;
+using DFC.ServiceTaxonomy.ApiFunction.Models;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Web.Http;
+using ILogger = Microsoft.Extensions.Logging.ILogger;
+
+namespace DFC.ServiceTaxonomy.ApiFunction.Function
+{
+    public class Execute
+    {
+        private readonly IOptions<ServiceTaxonomyApiSettings> _serviceTaxonomyApiSettings;
+        private readonly IHttpRequestHelper _httpRequestHelper;
+        private readonly IJsonHelper _jsonHelper;
+        private readonly INeo4JHelper _neo4JHelper;
+        private readonly IFileHelper _fileHelper;
+
+        public Execute(IOptions<ServiceTaxonomyApiSettings> serviceTaxonomyApiSettings, IHttpRequestHelper httpRequestHelper, IJsonHelper jsonHelper, INeo4JHelper neo4JHelper, IFileHelper fileHelper)
+        {
+            _serviceTaxonomyApiSettings = serviceTaxonomyApiSettings ?? 
+                                          throw new ArgumentNullException(nameof(serviceTaxonomyApiSettings));
+
+            _httpRequestHelper = httpRequestHelper ??
+                                 throw new ArgumentNullException(nameof(httpRequestHelper));
+
+            _jsonHelper = jsonHelper ??
+                  throw new ArgumentNullException(nameof(jsonHelper));
+
+            _neo4JHelper = neo4JHelper ?? 
+                           throw new ArgumentNullException(nameof(neo4JHelper));
+
+            _fileHelper = fileHelper ??
+                          throw new ArgumentNullException(nameof(fileHelper));
+        }
+
+        [FunctionName("Execute")]
+        public async Task<IActionResult> Run(
+            [HttpTrigger(AuthorizationLevel.Anonymous, "get", "post", Route = null)] HttpRequest req,
+            ILogger log)
+        {
+            var functionToProcess = _serviceTaxonomyApiSettings.Value.Function;
```

Comment:
I think we should wrap the top level function code in a try/catch block, so that we can capture, log and gracefully handle any unexpected exceptions.